main.py

```diff
@@ -15,6 +18,12 @@
 LOG_NAME = pathlib.Path(LOG_DIRECTORY_PATH, "image")
 DATA_COLLECTION_DELAY_TIME = 1.0  # seconds
 
+IMAGE_ENCODE_EXT = ".png"
+
+SOCKET_CREATION_MAX_ATTEMPTS = 10
```

Comment:
Is the socket really that unreliable?

Reply:
I'm not sure, I added this just as a safeguard.

Reply:
I see. When you test it let us know.